spec/rubocop/cop/rails/index_true_spec.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+describe RuboCop::Cop::Rails::IndexTrue, :config do
+  subject(:cop) { described_class.new(config) }
+
+  let(:source) do
+    <<-RUBY
+      class ExampleMigration < ActiveRecord::Migration
+        def change
+          #{code}
+        end
+      end
+    RUBY
+  end
+
+  shared_examples :accepts do |name, code|
+    let(:code) { code }
+
+    it "accepts usages of #{name}" do
+      inspect_source(cop, source)
+
+      expect(cop.offenses).to be_empty
+    end
+  end
+
+  shared_examples :offense do |name, code|
+    let(:code) { code }
+
+    it "registers an offense for #{name}" do
+      inspect_source(cop, source)
+
+      expect(cop.messages.any?).to eq(true)
+    end
+  end
+
+  it_behaves_like :accepts, 'add_column without index: true', <<-RUBY
+    add_column :books, :author_id, :integer
+  RUBY
+
+  it_behaves_like :offense, 'add_column with index: true', <<-RUBY
+    add_column :books, :author_id, :integer, index: true
+  RUBY
+
+  it_behaves_like :accepts, 'change_column without index: true', <<-RUBY
+    change_column :books, :author_id, :integer
+  RUBY
+
+  it_behaves_like :offense, 'change_column with index: true', <<-RUBY
+    add_column :books, :author_id, :integer, index: true
+  RUBY
+
+end
```

Comment:
we can use `expect_offense` and `expect_no_offenses` helpers for test.

For example: 

```
it '...' do
  expect_offense(<<-END.strip_indent)
    add_column :books, :author_id, :integer, index: true
    ^^^ MESSAGE
  END
end
```


See. https://github.com/bbatsov/rubocop/issues/3905